frontend/src/views/pages/UploadPage/UploadPage.tsx

```diff
@@ -39,13 +49,19 @@ function UploadPage() {
             linkRef.current?.reportValidity()
         } else if (file) {
             setPage(2)
+        } else if (link) {
+            if (link.trim() != "") {
+                s3_controller.handleS3Link(link);
```

Comment:
The controller call needs to be moved here (View Results button) so that it can switch to the Select Attribute page. 
Rn it directly processes the s3 bucket link instead of switching to attribute selection
<img width="939" alt="image" src="https://github.com/user-attachments/assets/7f612a36-4957-4f0c-8fe6-1fa7dccbfe5b">